src/interfaces/simulation.ts

```diff
@@ -38,84 +44,145 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
-      };
+    return { address, decimals, pricePerShare };
+  }
+
+  private async directDeposit({
+    depositProps: { toVault, amount, from, sellToken },
+    vaultContract,
+    options,
+  }: {
+    depositProps: DepositProps;
+    vaultContract: VaultContract;
+    options: SimulationOptions;
+  }): Promise<TransactionOutcome> {
+    const encodedInputData = await (this.shouldUsePartnerService(toVault)
+      ? this.yearn.services.partner?.encodeDeposit(toVault, amount)
+      : vaultContract.encodeDeposit(amount));
+
+    if (!encodedInputData) {
+      throw new Error("directDeposit#encodeDeposit failed");
     }
-    return this.simulationExecutor.executeSimulationWithReSimulationOnFailure(simulateDeposit, forkId);
+
+    const partnerAddress = await this.yearn.services.partner?.address;
+    const addressToDeposit = (this.shouldUsePartnerService(toVault) && partnerAddress) || toVault;
+
+    const tokensReceived = await this.simulationExecutor.simulateVaultInteraction(
+      from,
+      addressToDeposit,
+      encodedInputData,
+      toVault,
+      options
+    );
+
+    const targetTokenAmountUsdc = await this.yearn.services.oracle
+      .getNormalizedValueUsdc(toVault, tokensReceived)
+      .catch(() => {
+        throw new PriceFetchingError("Error fetching price", PriceFetchingError.FETCHING_PRICE_ORACLE);
+      });
+
+    const [decimals, pricePerShare] = await Promise.all([
+      vaultContract.decimals().catch(() => {
+        throw new EthersError("no decimals", EthersError.NO_DECIMALS);
+      }),
+      vaultContract.pricePerShare().catch(() => {
+        throw new EthersError("no price per share", EthersError.NO_PRICE_PER_SHARE);
+      }),
+    ]);
+
+    const targetUnderlyingTokensReceived = toBN(tokensReceived)
+      .div(toBN(10).pow(decimals))
+      .multipliedBy(pricePerShare)
+      .toFixed(0);
+
+    return {
+      sourceTokenAddress: sellToken,
+      sourceTokenAmount: amount,
+      targetTokenAddress: toVault,
+      targetTokenAmount: tokensReceived,
+      targetTokenAmountUsdc: targetTokenAmountUsdc,
+      targetUnderlyingTokenAddress: toVault,
+      targetUnderlyingTokenAmount: targetUnderlyingTokensReceived,
+      conversionRate: 1,
+      slippage: 0,
+    };
+  }
+
+  private async depositNeedsApproving({
+    depositProps: { from, sellToken, amount, toVault },
+    signer,
+  }: {
+    depositProps: DepositProps;
+    signer: JsonRpcSigner;
+  }): Promise<boolean> {
+    const TokenAbi = ["function allowance(address owner, address spender) view returns (uint256)"];
+    const contract = new Contract(sellToken, TokenAbi, signer);
+    const isUsingPartnerService = this.shouldUsePartnerService(toVault);
+    const addressToCheck = (isUsingPartnerService && this.yearn.services.partner?.address) || toVault;
+    const result = await contract.allowance(from, addressToCheck).catch(() => {
+      "deposit needs approving";
```

Comment:
this should throw an error?

Reply:
Again, existing logic, throw makes sense to me.

Throwing with https://github.com/yearn/yearn-sdk/pull/280/commits/03d391e9e80a1799b45af24e8b85d613358cb787